src/client/src/Client.c

```diff
@@ -1306,13 +1479,33 @@ STATUS freeKinesisVideoClientInternal(PKinesisVideoClient pKinesisVideoClient, S
         }
     }
 
+    // unLock the streamListLock after iteration
+    if (IS_VALID_MUTEX_VALUE(pKinesisVideoClient->base.streamListLock)) {

```

Comment:
here, you can check whether it's locked. If it is then it must have been valid too to be locked in the first place

Reply:
good point I'll make that change